admin/spec/back-end/controllers/access-arrangements.spec.js

```diff
@@ -335,16 +336,13 @@ describe('access arrangements controller:', () => {
         }
       }
     }
-    it('throws an error if edit mode not available', async () => {
+    it('redirects to error page if edit mode not available', async () => {
       spyOn(accessArrangementsService, 'getCurrentViewMode').and.returnValue(aaViewModes.readonly)
       const req = getReq(reqParams)
       const res = getRes()
-      try {
-        await controller.getDeleteAccessArrangements(req, res, next)
-        fail('error should have been thrown to prevent edit')
-      } catch (error) {
-        expect(error.name).toBe('AccessArrangementsNotEditableError')
-      }
+      await controller.getDeleteAccessArrangements(req, res, next)
+      expect(next).toHaveBeenCalledWith(new AccessArrangementsNotEditableError())
+      // TODO assert error page shown
```

Comment:
Is this TODO item for a later PR?

Reply:
no, that TODO has been resolved. will remove the comment